src/domain/ecoverse/ecoverse.service.ts

```diff
@@ -229,4 +232,21 @@ export class EcoverseService {
 
     return ctVerse;
   }
+
+  async createUser(userData: UserInput): Promise<IUser> {
```

Comment:
createUser should be in UserService, not EcoverseService

Reply:
this would create a circular reference as the user also needs to be added in to the members group - see pending issue. In general design has that you create the entity on the containing entite e.g. createGroup on org, challenge + ecoverse. A user is not standalone.

Reply:
No entity is standalone, but the user business logic should be in UserService. Circular modular / service reference can be resolved with forwardRef on module and service level (see latest pushed commits on server-159).

Reply:
In general every entity can be considered part of the Ecoverse, but that doesn't mean its business logic is part of the EcoverseService logic. It should be in its own service, following [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) principle.